### PR TITLE
Filter transactional columns and support projection for get/scan in Consensus Commit

### PR DESCRIPTION
**Pull request by brfrn169**
Currently, Get/Scan operations results include the transactional columns and non-projected columns in Consensus Commit. After this change, we will get the Get/Scan operations results without the transactional columns, and it also adds a Projection function support. Please take a look!

### CODE REVIEW

**Review by thongdk8 — DISMISSED** [review]
LGTM! Thank you!

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/FilteredResult.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+  private final Result result;
+  private final Set<String> projections;
+  private final Supplier<Map<String, Value<?>>> values;
+
+  public FilteredResult(Result result, List<String> projections) {
+    // assume that all the values are projected to the original result
+    this.result = Objects.requireNonNull(result);
+    this.projections = new HashSet<>(Objects.requireNonNull(projections));
+    values = Suppliers.memoize(this::filterValues); // lazy loading
```
Comment (feeblefakie) [inline core/src/main/java/com/scalar/db/transaction/consensuscommit/FilteredResult.java:33]: I didn't know this! Looks pretty useful.

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/FilteredResult.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** An implementation of {@code Result} to filter projected columns and transactional columns. */
```
Comment (feeblefakie) [inline core/src/main/java/com/scalar/db/transaction/consensuscommit/FilteredResult.java:21]:
```suggestion
/** An implementation of {@code Result} to filter out unprojected columns and transactional columns. */
```

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
Left one minor English suggestion.

**Pull request merged**